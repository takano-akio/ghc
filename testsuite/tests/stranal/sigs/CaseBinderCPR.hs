module CaseBinderCPR where

-- This example, taken from nofib's transform (and heavily reduced) ensures that
-- CPR information is added to a case binder

f_list_cmp::(t1 -> t1 -> Int) -> [t1] -> [t1] -> Int;
f_list_cmp a_cmp [] []= 0
f_list_cmp a_cmp [] a_ys= -1
f_list_cmp a_cmp a_xs []= 1
f_list_cmp a_cmp (a_x:a_xs) (a_y:a_ys)=
    if r_order == 0
    then f_list_cmp a_cmp a_xs a_ys
    else r_order
  where
    r_order = a_cmp a_x a_y
