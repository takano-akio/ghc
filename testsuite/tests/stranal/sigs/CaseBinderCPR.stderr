
==================== Strictness signatures ====================
CaseBinderCPR.$trModule: tm(tm1,tm1)
CaseBinderCPR.f_list_cmp: <L,C(C1(U(U)))><S,1*U><S,1*U>m()



==================== Strictness signatures ====================
CaseBinderCPR.$trModule: tm(tm1,tm1)
CaseBinderCPR.f_list_cmp: <L,C(C1(U(U)))><S,1*U><S,1*U>m(t)


